test/end2end_test.go

```diff
@@ -1398,3 +1441,98 @@ func leakCheck(t testing.TB) func() {
 		}
 	}
 }
+
+type lockingWriter struct {
+	mu sync.Mutex
+	w  io.Writer
+}
+
+func (lw *lockingWriter) Write(p []byte) (n int, err error) {
+	lw.mu.Lock()
+	defer lw.mu.Unlock()
+	return lw.w.Write(p)
+}
+
+func (lw *lockingWriter) setWriter(w io.Writer) {
+	lw.mu.Lock()
+	defer lw.mu.Unlock()
+	lw.w = w
+}
+
+var testLogOutput = &lockingWriter{w: os.Stderr}
+
+func awaitLogOutput(t *testing.T, maxWait time.Duration, phrase string) {
```

Comment:
Oh, yeah, I changed my mind about something earlier. Removed.

I have grander plans for all this logging stuff, but this is a baby step.